x-pack/plugins/observability_ai_assistant/public/components/chat/chat_body.tsx

```diff
@@ -104,6 +106,7 @@ export function ChatBody({
   knowledgeBase: UseKnowledgeBaseResult;
   currentUser?: Pick<AuthenticatedUser, 'full_name' | 'username'>;
   startedFrom?: StartedFrom;
+  chatFlyoutSecondSlotHandler?: ChatFlyoutSecondSlotHandler;
```

Comment:
Is there any way we can get rid of this? Can it be handled via actions as well, similar to visualize query?

Reply:
I don't like it either Dario, this is why I created this issue in the EUI repo https://github.com/elastic/eui/issues/7443. If the 2 types of flyouts could work together all this logic would just go away.

I need to move the secondSlot container from the flyout which holds this 2 slots logic to the `visualize_esql` function for the inline editing to be rendered correctly. I don't think we can use uiActions here unless I am missing something. 

I had also explored the idea to pass it on the `visualize_query` arguments but it won't pass due to JSON.stringify. Happy to explore other ideas though if you have something better in mind

Reply:
I believe it's good enough for now. Lets merge it and see if we can improve it later. An Eui solution would be preferable.

Reply:
I'm fine with a follow-up, but let's get rid of it regardless of what EUI does. We can use React context to stack modals